### ISSUE TITLE
[TRACKING] Add documentation

### ISSUE THREAD

**Issue by joelhawksley** [primer/view_components#77]
### Problem

Primer ViewComponents has little documentation. It's difficult for consumers of the library to know how to use it correctly. 

Developers looking for documentation expect to find it at [primer.style](https://primer.style), but their search comes up empty.

### Solution

Add a Primer ViewComponents section to [primer.style](https://primer.style).

### TODO

- [x] Test YARD documentation tool (done, see https://github.com/primer/view_components/pull/64).
- [x] Build out end-to-end proof of concept for generating markdown documentation from YARD, served by Doctocat.
- [x] Write documentation for all components, using [yardstick](https://github.com/dkubb/yardstick) to ensure complete coverage.
- [x] Ship documentation site on `primer.style`
- [x] Add Status field for each component like we have in Primer CSS (New, Stable, etc)

**Comment by joelhawksley** [comment]
Today, @broccolini, @colebemis, @yaili and I chatted about this project. Here are some notes:

**TL;DR**
We're going to proceed with the current approach (using YARD) to see how documenting more components goes.

**Editor integration**
@broccolini asked if YARD comments will allow for editor integrations, such as in VS Code. It looks like the [solargraph]
(https://marketplace.visualstudio.com/items?itemName=castwide.solargraph) plugin uses YARD comments for this kind of behavior 🎊 

**Open questions**
- How will we build out new components in Primer CSS alongside their ViewComponent equivalents?

**Comment by joelhawksley** [comment]
Per @kevinsawicki [here](https://github.slack.com/archives/C0ZCGGGJ2/p1604610411426400?thread_ts=1604610137.423300&cid=C0ZCGGGJ2), let's look at adding an explanation of the Octicon Size values 😄

**Comment by kevinsawicki** [comment]
Another idea, might be nice to have some small doc about the fundaments of slots. I get search results for them but not exactly what/why/how they are:


<img width="653" alt="Screen Shot 2020-11-05 at 1 12 39 PM" src="https://user-images.githubusercontent.com/671378/98296985-98781b00-1f68-11eb-9278-9a64ac6e0f56.png">

**Comment by joelhawksley** [comment]
Another idea: We should directly link to Storybook for each component.

**Comment by manuelpuyol** [comment]
@joelhawksley shall we close this?

**Comment by joelhawksley** [comment]
@manuelpuyol agreed!
